### PR TITLE
Adding initialisation list in c++ generator (-weffc++)

### PR DESCRIPTION
**Pull request by rherrmannr**

### CODE REVIEW

File: plugins/org.yakindu.sct.simulation.core/xtend-gen/org/yakindu/sct/simulation/core/util/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/.ExecutionContextExtensions.java._trace
+/.ExecutionContextExtensions.xtendbin
+/ExecutionContextExtensions.java
```
Comment (BeckmaR) [inline plugins/org.yakindu.sct.simulation.core/xtend-gen/org/yakindu/sct/simulation/core/util/.gitignore]: You should just exclude `/xtend-gen/` :)

**Comment by rherrmannr** [comment]
fix #1633

File: plugins/org.yakindu.sct.generator.cpp/src/org/yakindu/sct/generator/cpp/StatemachineHeader.xtend [hunk 2]
```diff
@@ -125,7 +125,7 @@ class StatemachineHeader extends org.yakindu.sct.generator.c.StatemachineHeader
 		'''
 	}
 	
-	def protected generatePrivateClassmembers(ExecutionFlow it) {
+		def protected generatePrivateClassmembers(ExecutionFlow it) {
```
Comment (terfloth) [inline plugins/org.yakindu.sct.generator.cpp/src/org/yakindu/sct/generator/cpp/StatemachineHeader.xtend]: reformat here please - indention looks wired

File: plugins/org.yakindu.sct.generator.cpp/src/org/yakindu/sct/generator/cpp/StatemachineImplementation.xtend [hunk 3]
```diff
@@ -99,25 +102,30 @@ class StatemachineImplementation implements IContentTemplate {
...
+			stateConfVectorPosition(0)«FOR s : getInterfaces»,
+			«s.instance»()«IF s.hasOperations && !entry.useStaticOPC»,
+			«s.OCB_Instance»(null)«ENDIF»«ENDFOR»
+		'''
+	}
+	
 	protected def CharSequence constructorBody(ExecutionFlow it)
 		'''
-		«scopes.filter(typeof(StatechartScope)).filter[hasOperations && !entry.useStaticOPC].map['''«OCB_Instance» = null;'''].join('\n')»
+«««		«scopes.filter(typeof(StatechartScope)).filter[hasOperations && !entry.useStaticOPC].map['''«OCB_Instance» = null;'''].join('\n')»
```
Comment (terfloth) [inline plugins/org.yakindu.sct.generator.cpp/src/org/yakindu/sct/generator/cpp/StatemachineImplementation.xtend]: is the ``«««`` at the beginning of the like correct?

File: plugins/org.yakindu.sct.simulation.core/xtend-gen/org/yakindu/sct/simulation/core/util/.gitignore (hunk 1 above)
Comment (terfloth) [reply plugins/org.yakindu.sct.simulation.core/xtend-gen/org/yakindu/sct/simulation/core/util/.gitignore]: Yes please ...

**Pull request merged**